src/MonoTorrent/MonoTorrent.Client.Messages/MessageBundle.cs

```diff
@@ -85,5 +86,10 @@ public override int Encode (byte[] buffer, int offset)
 
             return CheckWritten (written - offset);
         }
+
+        public override string ToString()
+        {
+            return $"MessageBundle [{String.Join(", ", this.Messages.Select(m => m.ToString()).ToArray())}]";
```

Comment:
Do you want `ToString` to aid debugging? If so you could implement it similar to how BitField does it ( https://github.com/alanmcgovern/monotorrent/blob/master/src/MonoTorrent/MonoTorrent/BitField.cs#L43 )